src/mad_spark_alt/core/smart_orchestrator.py

```
@@ -484,27 +484,68 @@ async def run_parallel_generation(
+                        logger.error(f"Error retrieving completed result: {e}")
+
+                # Cancel pending tasks to clean up resources
+                for task in pending:
+                    task.cancel()
+
+                # Wait briefly for cancellation to complete
+                await asyncio.gather(*pending, return_exceptions=True)
+
+                # Merge completed results with timeout placeholders
+                result_dict = {}
+                task_list = list(done) + list(pending)
```

Ruff (0.11.9)

Local variable `task_list` is assigned to but never used

Remove assignment to unused variable `task_list`

(F841)